LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/JsonHandlers/LnsStationConfiguration.cs

```diff
@@ -160,39 +160,57 @@ private class Sx1301Config
+                JsonReader.Array(from e in JsonReader.Tuple(JsonReader.Either(from n in JsonReader.UInt32().Validate(n => n == 0)
+                                                                              select FskSpreadingFactor,
+                                                                              JsonReader.UInt32().AsEnum(n => unchecked((SpreadingFactor)n))),
                                                                               from n in JsonReader.UInt32()
                                                                               select unchecked((Bandwidth)n),
                                                                               from n in JsonReader.UInt32()
                                                                               select n > 0)
-                                                   select (SpreadingFactor: e.Item1, Bandwidth: e.Item2, DownloadOnly: e.Item3)
-                                                   into e
-                                                   select e.SpreadingFactor is FskSpreadingFactor ? (FskSpreadingFactor, 0, e.DownloadOnly)
-                                                        : Enum.IsDefined(e.Bandwidth) ? (e.SpreadingFactor, e.Bandwidth, e.DownloadOnly)
-                                                        : throw new JsonException($"Invalid bandwidth: {e.Bandwidth}"))),
-                              JsonReader.Property("sx1301_conf", JsonReader.Array(Sx1301ConfReader)),
-                              JsonReader.Property("nocca", JsonReader.Boolean()),
-                              JsonReader.Property("nodc", JsonReader.Boolean()),
-                              JsonReader.Property("nodwell", JsonReader.Boolean()),
-                              (netId, joinEui, region, hwspec, freqRange, drs, sx1301conf, nocca, nodc, nodwell) =>
-                                    WriteRouterConfig(netId, joinEui, region, hwspec, freqRange, drs,
-                                                      sx1301conf, nocca, nodc, nodwell));
+                                 select (SpreadingFactor: e.Item1, Bandwidth: e.Item2, DownloadOnly: e.Item3)
+                                 into e
+                                 select e.SpreadingFactor is FskSpreadingFactor ? (FskSpreadingFactor, 0, e.DownloadOnly)
+                                     : Enum.IsDefined(e.Bandwidth) ? (e.SpreadingFactor, e.Bandwidth, e.DownloadOnly)
+                                     : throw new JsonException($"Invalid bandwidth: {e.Bandwidth}")));
+            var sx1301Reader = JsonReader.Property("sx1301_conf", JsonReader.Array(Sx1301ConfReader));
+            var noccaReader = JsonReader.Property("nocca", JsonReader.Boolean());
+            var nodcReader = JsonReader.Property("nodc", JsonReader.Boolean());
+            var nodwellReader = JsonReader.Property("nodwell", JsonReader.Boolean());
+            var bcningReader = JsonReader.Property("bcning",
```

Comment:
What if "bcning" is not there in the Twin? Isn't this going to fail as it is expecting it to be there?
Shouldn't we use the JsonReader.Either?